internal/upgradecheck/http_test.go

```diff
@@ -72,7 +72,7 @@ func TestCheckForUpgrades(t *testing.T) {
 		}
 
 		res, header, err := checkForUpgrades(ctx, "", "4.7.3", backoff,
-			fakeClient, cfg, false)
+			fakeClient, cfg, false, "")
 		assert.NilError(t, err)
 		assert.Equal(t, res, `{"pgo_versions":[{"tag":"v5.0.4"},{"tag":"v5.0.3"},{"tag":"v5.0.2"},{"tag":"v5.0.1"},{"tag":"v5.0.0"}]}`)
 		checkData(t, header)
```

Comment:
Should `checkData` be updated to check for the registration token? Should there be a case where we pass a non-empty string for the token and check that it makes it into the header properly?